### PR TITLE
Move network

### PR DESCRIPTION
**Pull request by nicboul**

### CODE REVIEW

File: network.go [hunk 1]
```diff
@@ -0,0 +1,765 @@
...
+}
+
+var gMAC []byte
+var gSwitch switchInstance
+var utun *water.Interface
+var vswitchConn *tls.Conn
+
+var (
+	// arpQueue of waiting frames that is global to main.
+	arpQueue *ARPQueue
```
Comment (si74) [inline network.go:85]: I think this needs to be exported if used outside of package agent - unless it won't be?

File: network.go [hunk 2]
```diff
@@ -0,0 +1,765 @@
...
+		if err != nil {
+			Lerror.Printf("failed to read from %s: %v\n", utunName, err)
+		}
+		Ldebug.Printf("read %d bytes from %s\n", n, utunName)
+
+		if vswitchConn == nil {
+			continue
+		}
+
+		// nvHeader lenght value
```
Comment (si74) [inline network.go:138]: nit: "length"

File: network.go [hunk 3]
```diff
@@ -0,0 +1,765 @@
...
+
+	config := water.Config{
+		DeviceType: water.TUN,
+	}
+	config.Name = utunName
+
+	var err error
+	utun, err = water.New(config)
+	if err != nil {
+		Lerror.Fatalf("failed to initialize the %s interface: %v\n", utunName, err)
```
Comment (si74) [inline network.go:763]: I would actually return an error and not run readUTun in main.go above if this fails to initialize

Comment (nicboul) [reply network.go:763]: agreed, will update in the next PR

File: network.go (hunk 1 above)
Comment (nicboul) [reply network.go:85]: hmm good question, I'll check this one once I start using it 🤔

**Pull request merged**